Core/Src/error_treatment.c

```diff
@@ -6,19 +6,22 @@
  */
 
 #include "error_treatment.h"
-#include "cmsis_os.h"
 #include "global_instances.h"
 
 
-void check_for_errors(bool (* areThereErrors)(), int flagError) {
-    if (areThereErrors()) {
-        issue_error(flagError);                             // chama o erro para a main_task tratar
-        osEventFlagsSet(ECU_control_event_id, flagError);   // seta flag de estado com a flag flagError
-    } else {
-        osEventFlagsClear(ECU_control_event_id, flagError); // limpa flag de estado flagError
-    }
+void check_for_errors(bool (* areThereErrors)(), uint32_t flagError) {
+    if (areThereErrors())
+        issue_error(flagError, true);   // seta flag de estado com a flag flagError
```

Comment:
```suggestion
        issue_error(flagError, /*should_set_control_event_flag=*/true);   // seta flag de estado com a flag flagError
```
sempre faz isso quando vc chamar uma função com um literal, assim a gente sabe o que aquele `true` significa